roboteam_ai/src/world/Ball.h

```diff
@@ -39,7 +39,7 @@ class Ball {
         Ball();
         // Ball(const Ball &copy) = default;
         explicit Ball(const roboteam_msgs::WorldBall &copy);
-        void updateBall(const BallPtr &oldBall, const WorldData &worldData);
+        void updateBall(const BallPtr &oldBall, const WorldData &worldData, bool applyBallFilter = true);
```

Comment:
This is not the way to do it. What if you don't call the filter for some time and then reset it again? We either use it or we do not.

Reply:
the filter is normally always applied, but it breaks some tests which use linear extrapolation of the ball velocity